src/tainting/Taint.ml

```diff
@@ -114,6 +115,8 @@ and compare_sources s1 s2 =
   let pm1, ts1 = pm_of_trace s1.call_trace
   and pm2, ts2 = pm_of_trace s2.call_trace in
   match
+    (* TODO: I'm pretty suspicious of Stdlib.compare here,
+       the metavariable environments include tokens *)
```

Comment:
Totally agree.

Reply:
I'm really considering hand-writting almost every compare.

Reply:
I'm going to handwrite this compare today and see what happens